internal/devpkg/pkgtype/flake.go

```diff
@@ -0,0 +1,20 @@
+package pkgtype
+
+import (
+	"go.jetpack.io/devbox/nix/flake"
+	"strings"
+)
+
+func IsFlake(s string) bool {
```

Comment:
I think you can reuse `devpkg.isAmbiguous` here.

In some cases it's impossible to know if a package is a flake without resolving it (for example, `foo` could be an indirect flake's default package or it could be a Devbox package named `foo`).

Reply:
A couple issue here:
1. devpkg.isAmbiguous would need to have it's visibility changed
2. devpkg.isAmbiguous would need to move to a different package because using devpkg in the lock package causes an import cycle

Reply:
@gcurtis One issue before this PR is that `resolve.FetchResolvedPackage` would call `searcher.ParseVersionedPackage` and that would incorrectly identify `git+ssh://git@foo/bar/baz.git#boo` as a versioned package due to the `@`.  Filtering out known Flakes and leaving ambiguous Flakes is probably good enough at this point.

Reply:
I switched to using `ParseInstallable` and `IsAmbiguous` in `pkgtype.IsFlake`.  To do so, I moved `IsAmbiguous` to package `pkgtype`.